fsm/src/main/java/org/egov/fsm/repository/rowmapper/FSMRowMapper.java

```diff
@@ -0,0 +1,118 @@
+import org.egov.fsm.web.model.PitDetail;
+import org.egov.fsm.web.model.location.Address;
+import org.egov.fsm.web.model.location.Boundary;
+import org.egov.fsm.web.model.location.GeoLocation;
+import org.postgresql.util.PGobject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.ResultSetExtractor;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+
+@Component
+public class FSMRowMapper implements ResultSetExtractor<List<FSM>> {
+
+	@Autowired
+	private ObjectMapper mapper;
+
+	@SuppressWarnings("rawtypes")
+	@Override
+	public List<FSM> extractData(ResultSet rs) throws SQLException, DataAccessException {
+
+		Map<String, FSM> fmsMap = new LinkedHashMap<String, FSM>();
+
+		while (rs.next()) {
+			FSM currentfsm = new FSM();
+			// TODO fill the FSM object with data in the result set record
+			String id = rs.getString("fsm_id");
+			String applicationNo = rs.getString("applicationno");
+			currentfsm = fmsMap.get(id);
+			String tenantId = rs.getString("tenantid");
+			String accountId = rs.getString("accountId");
+			String description = rs.getString("description");
+			String additionalDetails = rs.getString("additionalDetails");
+			String source = rs.getString("source");
+			String sanitationtype = rs.getString("sanitationtype");
+			String propertyUsage = rs.getString("propertyUsage");
+			int noOfTrips = rs.getInt("noOfTrips");
+			String applicationStatus = rs.getString("applicationStatus");
+			String status = rs.getString("status");
+			String vehicleId = rs.getString("vehicle_id");
+			if (currentfsm == null) {
+				Long lastModifiedTime = rs.getLong("lastmodifiedtime");
+
+				if (rs.wasNull()) {
+					lastModifiedTime = null;
+				}
+				currentfsm = FSM.builder().id(id).applicationNo(applicationNo).tenantId(tenantId)
+						.description(description).accountId(accountId).additionalDetails(additionalDetails)
+						.source(source).sanitationtype(sanitationtype).propertyUsage(propertyUsage).noOfTrips(noOfTrips)
+						.vehicleId(vehicleId).applicationStatus(applicationStatus)
+						.build();
+
+				fmsMap.put(id, currentfsm);
+			}
+			addChildrenToProperty(rs, currentfsm);
+
+		}
+
+		return new ArrayList<>(fmsMap.values());
+
+	}
+
+	@SuppressWarnings("unused")
+	private void addChildrenToProperty(ResultSet rs, FSM fsm) throws SQLException {
+
+		// TODO add all the child data Vehicle, Pit, address
+		String tenantId = fsm.getTenantId(); 
+		
+		AuditDetails auditdetails = AuditDetails.builder().createdBy(rs.getString("createdBy"))
+				.createdTime(rs.getLong("createdTime")).lastModifiedBy(rs.getString("lastModifiedBy"))
+				.lastModifiedTime(rs.getLong("lastModifiedTime")).build();
+
+		Double latitude = (Double) rs.getObject("latitude");

```

Comment:
if latitude or longitude is null in DB, it will be set to zero here and zero will be returned instead of null. Please check other municipal service rowmapper, we have handled this use case there

Reply:
i think instead of null it should return zero. if this returns null UI might have problem in rendering..
i have verified TL, PT and PGR could not find any solution..
please advise.